core/src/main/java/com/datastrato/gravitino/StringIdentifier.java

```diff
@@ -108,6 +109,27 @@ public static Map<String, String> addToProperties(
         .build();
   }
 
+  /**
+   * Remove StringIdentifier from properties.
+   *
+   * @param properties the properties to remove the string identifier from.
+   * @return the properties with the string identifier removed.
+   */
+  public static Map<String, String> removeFromProperties(Map<String, String> properties) {
+    if (properties == null) {
+      return null;
+    }
+
+    if (!properties.containsKey(ID_KEY)) {
+      return properties;
+    }
+
+    Map<String, String> copy = Maps.newHashMap(properties);
+    copy.remove(ID_KEY);
+
+    return ImmutableMap.copyOf(copy);
```

Comment:
This unnecessarily copies the map twice, you can optimize this logic to remove one copy.

Reply:
I want to return an immutable map here, if it does not matter, the last copy can be omitted.

Reply:
No, you can build the immutable map using `builder`, not by copying.